C7GameData/UnitPrototype.cs

```diff
@@ -21,5 +20,14 @@ public override string ToString()
         {
             return name + " (" + attack + "/" + defense + "/" + movement + ")";
```

Comment:
Some quick searching suggests this is optimized to be more efficient than it would in Java, but I think interpolation is the preferred method these days:
`$"{name} ({attack}/{defense}/{movement})`
Kind of a stylistic thing though. I find it more readable but open to suggestions.

Reply:
That makes sense.  I tend to default to the simple style because every language has its own syntax for the interpolated style, so I tend to forget which one is the correct one to use in the language I'm currently writing in, and spend more time getting sidetracked by that than it would take to write the non-interpolated version.  But it looks readable enough with interpolation in C#, even with the parentheses in this string making it harder to read than it would be in most cases.

Fun fact, string concatenation being slow in Java hasn't been true for about 15 years, at least outside of some edge cases.  It was either in Java 1.4 or 1.5 (the latter being released in 2004) when they added compiler optimizations to use the StringBuilder class under the hood (maybe StringBuffer if it was 1.4), which ended the performance problem of re-initializing a String for every item being +'ed together.  There are still a lot of old sites, or sites quoting old sites, that say string concatenation performance matters in Java, but in practice these days the right answer is "use whatever reads easier" 95% of the time.  The other 5% is typically when you're concatenating a _lot_ of strings for debug loggers, using an older logging framework, and not checking if debug logs are enabled before creating your strings.  Then it doesn't really matter how you're doing it, just that the volume can slow things down if it's e.g. printing a debug statement for every pixel of a 20 megapixel PCX file.

No idea if that applies to C# though, so it might make a more significant difference here.

I've changed this one and will try to remember this for future string interpolation events, now that I know the C# syntax for it.

Reply:
Good to know about Java! I haven't written it since (Android's approximation of) 7. In C# they seem to be pretty equivalent inline, but if you're doing dynamic concatenation like you mentioned you should use StringBuilder.

Reply:
Good to know that there's a StringBuilder in C# as well!